lib/git/lib.rb

```diff
@@ -789,6 +789,9 @@ def push(remote, branch = 'master', opts = {})
       arr_opts << '--mirror'  if opts[:mirror]
       arr_opts << '--delete'  if opts[:delete]
       arr_opts << '--force'  if opts[:force] || opts[:f]
+      if opts[:push_options]
+       arr_opts.concat(opts[:push_options].map { |push_option| ['--push-option', push_option] }.flatten)
+      end
```

Comment:
When I look at the rest of `Git::Lib`, I notice the mapping from `opts` to `arr_opts` where the `opts` value can multi-valued is inconsistently implemented. 

What do you think about providing a common way of doing multi-valued options:

```Ruby
def multi_valued_opt(opt, values)
  Array(values).map { |value| [opt, value] }.flatten
end
```

This implementation uses `Array(...)` so that the value for (for example) `opts[:push_options]` can be either a single object or an Array or any object that knows how to convert itself into an array.

and then implementing `--push-option` with this line:

```Ruby
arr_opts += multi_valued_opt('--push-option', opts[:push_options]) if opts[:push_options]
```

The hope would be that other instances of this pattern would adopt `multi_valued_opt` to make their code more understandable.